lib/parser/ruby27.y

```diff
@@ -2096,7 +2096,7 @@ opt_block_args_tail:
 
        p_var_ref: tCARET tIDENTIFIER
                     {
-                      lvar = @builder.accessible(@builder.ident(val[1]))
+                      lvar = @builder.accessible_for_pin(@builder.ident(val[1]))
```

Comment:
I think there's no need to create a separate method on a `Builder`. You have a local variable name in the `val[1]`, so you can do an inline check using `@static_env`